### PR TITLE
clean up json handling

### PR DESCRIPTION
**Pull request by macklin-10x**
Replaces all custom JSON parsing code with serde.  Inverts control flows to return values instead of mutating collections provided by arguments.

Stacked on another feature branch, will rebase after underlying branch merges.

### CODE REVIEW

File: enclone_args/src/read_json.rs [hunk 1]
```diff
@@ -175,126 +133,89 @@ fn parse_vector_entry_from_json(
...
-    // Read fraction_of_reads_for_this_barcode_provided_as_input_to_assembly.
-
-    let mut frac_reads_used = None;
-    let f = v["fraction_of_reads_for_this_barcode_provided_as_input_to_assembly"].as_f64();
-    if let Some(f) = f {
-        frac_reads_used = Some((f * 1_000_000.0).round() as u32);
-    }
+    let frac_reads_used = ann
+        .fraction_of_reads_for_this_barcode_provided_as_input_to_assembly
+        .map(|f| (f * 1_000_000.0).round() as u32);
```
Comment (sreenathkrishnan) [inline enclone_args/src/read_json.rs:139]: Wonder what is happening here with the 1M.

**Review by sreenathkrishnan — APPROVED** [review]
Much nicer!!

File: enclone_args/src/read_json.rs (hunk 1 above)
Comment (macklin-10x) [reply enclone_args/src/read_json.rs:139]: 🤷

File: enclone_args/src/read_json.rs [hunk 2]
```diff
@@ -715,83 +593,47 @@ pub fn read_json(
...
+                accept_inconsistent,
+                origin_info,
+                dataset_index,
+                refdata,
+                to_ref_index,
+                reannotate,
+                ctl,
+            )
+        })
+        .collect::<Result<Vec<_>, String>>()?;
```
[the remark this answers is not in the document]
Comment (macklin-10x) [reply enclone_args/src/read_json.rs:616]: Hmm, interesting idea, and I can't see any reason not to do this.  I'll make the change.

[the remark this answers is not in the document]
Comment (macklin-10x) [reply enclone_args/src/read_json.rs]: Done.

**Pull request merged**